src/services/variableHelpers.ts

```diff
@@ -1,38 +1,24 @@
+import { AdHocVariableFilter } from '@grafana/data';
 import { AdHocFiltersVariable, AdHocFilterWithLabels, sceneGraph, SceneObject } from '@grafana/scenes';
 
 import { IndexScene } from '../Components/IndexScene/IndexScene';
 import { ServiceScene } from '../Components/ServiceScene/ServiceScene';
+import { areArraysEqual } from './comparison';
 import { CustomConstantVariable } from './CustomConstantVariable';
 import { FilterOp } from './filterTypes';
 import { isOperatorInclusive } from './operatorHelpers';
 import { includeOperators, numericOperators, operators } from './operators';
-import { ReadOnlyAdHocFiltersVariable } from './ReadOnlyAdHocFiltersVariable';
 import { getRouteParams } from './routing';
 import { getLabelsVariable } from './variableGetters';
 import { SERVICE_NAME, SERVICE_UI_LABEL, VAR_LABELS } from './variables';
 
-type ClearableVariable = AdHocFiltersVariable | CustomConstantVariable | ReadOnlyAdHocFiltersVariable;
+type ClearableVariable = AdHocFiltersVariable | CustomConstantVariable;
 export function getVariablesThatCanBeCleared(indexScene: IndexScene): ClearableVariable[] {
   const variables = sceneGraph.getVariables(indexScene);
   let variablesToClear: ClearableVariable[] = [];
 
   for (const variable of variables.state.variables) {
-    if (variable instanceof ReadOnlyAdHocFiltersVariable) {
-      if (
-        !variable.state.filters.every((filter) =>
-          variable
-            .getReadonlyFilters()
-            ?.find(
-              (readonlyFilter) =>
-                readonlyFilter.key === filter.key &&
-                readonlyFilter.value === filter.value &&
-                readonlyFilter.operator === filter.operator
-            )
-        )
-      ) {
-        variablesToClear.push(variable);
-      }
-    } else if (variable instanceof AdHocFiltersVariable && variable.state.filters.length) {
+    if (variable instanceof AdHocFiltersVariable && variable.state.filters.length) {
```

Comment:
@ifrost I think this change is what's causing the initial labels to get removed (and most likely also a bug in my other branch removing the read only filters), we were relying on the readonly variable to let us know which filters should get left alone when clearing.